binding/Binding/SKPath.cs

```diff
@@ -480,102 +482,311 @@ public static int ConvertConicToQuads (SKPoint p0, SKPoint p1, SKPoint p2, float
 			}
 		}
 
-		public class Iterator : SKObject
+		private static void MorphPoints (
+					Span<SKPoint> dst,
+					Span<SKPoint> src,
+					int count,
+					SKPathMeasure meas,
+					in SKMatrix matrix)
 		{
-			private readonly SKPath path;
+			for (int i = 0; i < count; i++) {
+				SKPoint s = matrix.MapPoint (src[i].X, src[i].Y);
 
-			internal Iterator (SKPath path, bool forceClose)
-				: base (SkiaApi.sk_path_create_iter (path.Handle, forceClose ? 1 : 0), true)
-			{
-				this.path = path;
+				if (!meas.GetPositionAndTangent (s.X, out var p, out var t)) {
+					// set to 0 if the measure failed, so that we just set dst == pos
+					t = SKPoint.Empty;
+				}
+
+				// y-offset the point in the direction of the normal vector on the path.
+				dst[i] = new SKPoint (p.X - t.Y * s.Y, p.Y + t.X * s.Y);
+			}
+		}
+
+		/*  TODO: Need differentially more subdivisions when the follow-path is curvy. Not sure how to
+		 determine that, but we need it. I guess a cheap answer is let the caller tell us,
+		 but that seems like a cop-out. Another answer is to get Skia's Rob Johnson to figure it out.
+		 */
+		private static void MorphPath (SKPath dst, SKPath src, SKPathMeasure meas, in SKMatrix matrix)
+		{
+			using var it = src.CreateIterator (false);
+
+			SKPathVerb verb;
+
+			var srcP = new SKPoint[4];
+			var dstP = new SKPoint[4];
+
+			while ((verb = it.Next (srcP)) != SKPathVerb.Done) {
+				switch (verb) {
+					case SKPathVerb.Move:
+						MorphPoints (dstP, srcP, 1, meas, matrix);
+						dst.MoveTo (dstP[0]);
+						break;
+					case SKPathVerb.Line:
+						// turn lines into quads to look bendy
+						srcP[0].X = (srcP[0].X + srcP[1].X) * 0.5f;
+						srcP[0].Y = (srcP[0].Y + srcP[1].Y) * 0.5f;
+						MorphPoints (dstP, srcP, 2, meas, matrix);
+						dst.QuadTo (dstP[0], dstP[1]);
+						break;
+					case SKPathVerb.Quad:
+						MorphPoints (dstP, srcP.AsSpan ().Slice (1, 2), 2, meas, matrix);
+						dst.QuadTo (dstP[0], dstP[1]);
+						break;
+					case SKPathVerb.Conic:
+						MorphPoints (dstP, srcP.AsSpan ().Slice (1, 2), 2, meas, matrix);
+						dst.ConicTo (dstP[0], dstP[1], it.ConicWeight ());
+						break;
+					case SKPathVerb.Cubic:
+						MorphPoints (dstP, srcP.AsSpan ().Slice (1, 3), 3, meas, matrix);
+						dst.CubicTo (dstP[0], dstP[1], dstP[2]);
+						break;
+					case SKPathVerb.Close:
+						dst.Close ();
+						break;
+					default:
+						Debug.Fail ("unknown verb");
+						break;
+				}
 			}
+		}
+
+		public static SKPath CreateWarpedTextOnPath (
```

Comment:
Are these meant to be public? Also, if these are pretty "generic", then maybe we can make them a "feature" on `SKFont`? There is a `GetPath` method set there already... That method is new, so we can actually rename it to `GetTextPath` and then use these 2 new ones to `GetTextPathOnPath` or something? This might be a useful thing for the world.

Reply:
If possible, I would like these to be public, because they allow just getting the path or glyph blob, instead of drawing it.

Okay, I will move these to `SkFont`, good idea.

Reply:
I took a hybrid approach, moving the bulk of the code into separate classes, but making these extensions methods on `SKFont`. Otherwise the `SKFont` file would become really big with code that doesn't exist in the Skia API, so it felt more correct to keep it out of that file. Can be refactored  :)